tecken/base/symboldownloader.py

```diff
@@ -48,14 +50,40 @@ def wrapper(self, *args, **kwargs):
     miss_callable=lambda *a, **k: metrics.incr("symboldownloader_exists_cache_miss", 1),
 )
 @metrics.timer_decorator("symboldownloader_exists")
-def check_url_head(url):
+def get_last_modified(url: str) -> datetime.datetime:
+    """
+    Get the last modified date of the given URL.
+
+    This function performs a HEAD request to the given URL. If the status code is 200,
+    the Last-Modified header is parsed into a datetime.datetime object and returned.
+    If the response does not include a Last-Modified header, or the header can't be
+    parsed, the current time is returned. If the response status code is not 200, the
+    function returns None.
+
+    :arg url: The target URL.
+    :returns: The time the resource at the URL was last modified.
+    """
     session = session_with_retries(status_forcelist=(429, 500, 503))
     resp = session.head(url)
     # NOTE(willkg): we get a 403 from S3 buckets HTTP requests, so we want to ignore
     # those
     if resp.status_code not in (200, 403, 404):
-        logger.error(f"check_url_head: {url} status code is {resp.status_code}")
-    return resp.status_code == 200
+        logger.error(f"get_last_modified: {url} status code is {resp.status_code}")
+    if resp.status_code == 200:
+        try:
+            return parsedate_to_datetime(resp.headers["last-modified"])
+        except (ValueError, KeyError):
+            # KeyError occurs when the response does not hav a Last-Modified header,
+            # and ValueError occurs if the Last-Modified header isn't properly
+            # RFC-5322-formatted. Neither of this should ever happen, since S3 always
+            # includes a properly formatted Last-Modified header in responses, so
+            # this code is just a fallback to avoid erroring out if something
+            # unexpected happened.
+            logger.error(
+                f"get_last_modified: HEAD request to {url} did not return "
+                "a valid last-modified header"
+            )
```

Comment:
I thought about changing that, but then moved it to my general cleanups branch, since there are many instances of this in the codebase. I'll fix that single instance here, and wait with the other ones until after you have done your refactoring.